eng/testing/tests.android.targets

```diff
@@ -69,6 +69,8 @@
           DestinationFolder="$(TestArchiveTestsDir)"
           SkipUnchangedFiles="true"
           Condition="'$(ArchiveTests)' == 'true' and '$(IgnoreForCI)' != 'true'" />
+
+    <RemoveDir Condition="'$(ArchiveTests)' == 'true'" Directories="$(OutDir)" />
```

Comment:
The target is conditioned under ArchiveTests, we wouldn't need this condition right?
I saw we are using `NeedsToBuildAppsOnHelixLocal` on the ioslike counterpart, but also didn't see any occurrences of that in this repo, is that something we used to have or is it declared somewhere else? If it is populated through somewhere else maybe we can use that same condition?

Reply:
`NeedsToBuildAppsOnHelixLocal` is a condition where it won't delete when we are building an ios test on helix. Since we don't do that on android, it's not necessary to keep a similar condition.  I think the `ArchiveTests` check makes more sense now since that's the condition CI uses to build all of the tests.